journal/202309.md

```diff
@@ -0,0 +1,971 @@
+
+- Restored the ability to run [vspd on simnet](https://github.com/decred/vspd/pull/419), this is needed by DCRDEX to test new VSP staking features.
+- Small performance optimizations by making [better use of dcrd](https://github.com/decred/vspd/pull/422).
+- Unexported some code and [moved](https://github.com/decred/vspd/pull/428) it to [internal](https://github.com/decred/vspd/pull/430) [packages](https://github.com/decred/vspd/pull/436) since it is not suitable for consuming by third parties. Smaller public API surface makes development and maintenance easier.
+- Reworked how components are [created, started](https://github.com/decred/vspd/pull/434), and [stopped](https://github.com/decred/vspd/pull/436) to decouple and simplify code.
+- Fixed [database backup](https://github.com/decred/vspd/pull/439) running only once instead of running periodically (unreleased bug).
+- ~20 commits with smaller improvements and code cleanup.
+  - { _drop after review: 423 (spent scan fix), 433 (move), 429 (import kill), 432 (unparam), 431 (minor opti), 437 (rename), 441 (unused), 444 (x4 decomp, renames), 449 (deps), 446 (x8 minor cleanups and tweaks), 450 (ver)_ }
+
+{ _image candidates:_
+
+- new counters + banner
+  - https://github.com/decred/vspd/pull/417
+- missed page
+  - https://github.com/decred/vspd/pull/451
+
+}
+
+
+### gominer
+
+_[gominer](https://github.com/decred/gominer) { intro }_
+
+The switch to BLAKE3 mining has triggered a refresh of the gominer codebase after almost 3 years of little development activity. The main goal was to quickly react to the first GPU miners who did not share their code publicly and began to dominate the hashrate, and allow anyone mine with their GPUs. Besides adding BLAKE3 GPU mining support, effort has been made to automate or document the setup process to address multiple pain points reported by the miners.
+
+BLAKE3 GPU mining:
+
+- Added support for [BLAKE3 GPU mining via OpenCL](https://github.com/decred/gominer/pull/194). It includes a custom optimized BLAKE3 OpenCL kernel based on midstates for supporting GPUs that work with OpenCL and OpenCL with ADL. "Kernel" is a program written in a specialized [OpenCL kernel language](https://en.wikipedia.org/wiki/OpenCL#OpenCL_kernel_language) that runs on the GPU. This change also improves discovery of CL devices, randomizes nonces to ensure each device is doing different work, and improves displaying of hashrate values.
+  - { _was #194 prepared in advance? It came super fast after GPUs showed up!_ }
+  - { _when first private GPU miners were spotted?_ }
+- Added support for [BLAKE3 GPU mining with CUDA](https://github.com/decred/gominer/pull/195) on Linux. CUDA is a platform for high performance computing on GPUs. The OpenCL kernel has been adjusted to compile with the CUDA toolkit. Unlike OpenCL, [CUDA](https://en.wikipedia.org/wiki/CUDA) is proprietary, closed source, and only works on Nvidia cards. 
+  - { _who would use CUDA if OpenCL is faster?_ }
```

Comment:
CUDA allows extra functionality such as controlling overclocks and fan speeds to potentially significantly reduce power usage in exchange for very little difference hash rate.  Given the cost of electricity is a huge factor in profitability, such tradeoffs often makes sense.

Reply:
Added a slightly compressed version.